ISISScientificComputing/autoreduce
# Refactor ppa 'do not overwrite' handling

### Summary of work
- This PR refactors the code which handles cases where the overwrite flag is False/None in `post_process_admin.reduce`
- The old implementation was broken, but never used until changes to other areas of the code were changed (#557)
- _Note: to enable this code to run, additional changes were required._
  * Due to the Message class having been integrated into manual submission (#542), the data dictionary passed to PPA contained a full set of keys.
  * The previous code only checked whether a `message` key existed. This PR extended this by checking whether the value was `None`.
  * Additionally, since message cannot be null in the DB, this PR converts a None `message` to `=""`
    * This is an open question regarding where in the code this should occur.
___
- Additionally, this PR cherry-picks commits from #557 which revert the data_dict keys (/Message attributes) to their original names

### How to test your work
- Pull the code to the dev server
- Perform a manual submission
- Ensure the run completes without error

### Additional comments
 

Fixes #564


**Before merging ensure the release notes have been updated**
